### PR TITLE
Resolve bugs tied to record creations on table

### PR DESCRIPTION
**Pull request by charlesBochet**
In this PR, I'm resolving some bugs introduced earlier:
- When creating a new record on the table, exiting the inbox is deleting the record. This is buggy and as we are shifting to a record creation on Right drawer, I've removed this behavior
- Force record to be inserted in position 0
- Remove board create button for now as we don't have the possibility to select the column anymore

### CODE REVIEW

File: packages/twenty-front/src/pages/object-record/RecordIndexPage.tsx [hunk 1]
```diff
@@ -28,44 +24,29 @@ export const RecordIndexPage = () => {
...
   const setHotkeyScope = useSetHotkeyScope();
 
   const { setSelectedTableCellEditMode } = useSelectedTableCellEditMode({
     scopeId: recordIndexId,
   });
 
   const handleAddButtonClick = async () => {
-    await createOneObject?.({});
+    await createOneObject?.({
+      position: 0,
```
Comment (Weiko) [inline packages/twenty-front/src/pages/object-record/RecordIndexPage.tsx:40]: I'm assuming there is already one element at position 0 when you load the page right? Will it still work? @charlesBochet

File: packages/twenty-front/src/pages/object-record/RecordIndexPageHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  const Icon = getIcon(
+    findObjectMetadataItemByNamePlural(objectNamePlural)?.icon,
+  );
+
+  const recordIndexViewType = useRecoilValue(recordIndexViewTypeState);
+
+  return (
+    <PageHeader
+      title={
+        objectNamePlural.charAt(0).toUpperCase() + objectNamePlural.slice(1)
```
Comment (Weiko) [inline packages/twenty-front/src/pages/object-record/RecordIndexPageHeader.tsx:34]: @charlesBochet there is a capitalize util function in packages/twenty-front/src/utils/string/capitalize.ts

**Review by Weiko — APPROVED** [review]
Left comments but LGTM @charlesBochet !

File: packages/twenty-front/src/pages/object-record/RecordIndexPageHeader.tsx (hunk 2 above)
Comment (charlesBochet) [reply packages/twenty-front/src/pages/object-record/RecordIndexPageHeader.tsx:34]: I've split a file in two so I missed it, but I'll use capitalize, thank you for pointing it

File: packages/twenty-front/src/pages/object-record/RecordIndexPage.tsx (hunk 1 above)
Comment (charlesBochet) [reply packages/twenty-front/src/pages/object-record/RecordIndexPage.tsx:40]: It works but it's a hack. We will need to do a better job but this is fixing the current erratic component. I'll create a follow up ticket to properly implement the position that should be first / 2

[the remark this answers is not in the document]
Comment (charlesBochet) [reply packages/twenty-front/src/pages/object-record/RecordIndexPage.tsx:40]: @martmull this is actually tied to another behavior:
we should not create records with null position (which is happening when we create new workspaces or seed workspaces). 
For existing workspace, my plan is to update all existing one in DB for now
For workspace, prefill I have added it to the PR
For demo + seed dev, I did it last week

[the remark this answers is not in the document]
Comment (martmull) [reply packages/twenty-front/src/pages/object-record/RecordIndexPage.tsx:40]: ok then go

**Review by martmull — APPROVED** [review]

**Pull request merged**